### DISCUSSION TITLE
an image viewer/zoomer

### DISCUSSION THREAD

**Discussion by yassinebridi** [forem/forem#15895] [Feature Requests]
## Feature Request
an image viewer.
### User Story / Description
the platform has a really outdated way of viewing or zooming images, when you click an image, the page goes to the source of that image.
### Definition of Done
a simple performant way of zooming images, like medium has.

**Comment by jessleenyc** [comment]
Lightbox was suggested earlier: https://github.com/thepracticaldev/dev.to/issues/231

**Comment by benhalpern** [comment]
Cool, I don't share the concern about this feature as you do, but I'm okay with the implementation. We'll want a second "larger" version of the image than the one we display on the article. Currently we just link to the one you're looking at it and don't even zoom.

So I approve of this, but only if implemented the right way. Happy to provide support if someone wants to make it happen.

**Comment by stale** [comment]
Thanks for contributing to this issue. As it has been 90 days since the last activity, we are automatically closing the issue in 7 days. This is often because the request was already solved in some way and it just wasn't updated or it's no longer applicable. If this issue still requires attention, please respond with a comment. Happy Coding!

**Comment by deadcoder0904** [comment]
Yeah this is the only thing I hate about dev.to

It completely reloads the page again when you go back from the source. An overlay like Medium might be good :)

**Comment by decadef20** [comment]
There is PR about this issue? If not, maybe I can have a try.

**Comment by rhymes** [comment]
@decadef20 no, there isn't AFAIK, it's quite an old issue. Maybe @benhalpern has some ideas on this?

I'm not sure what the right implementation would be, I don't see much advantage into opening the same image in a modal/overlay, it also defeats the purpose of having an HREF I think. Media websites usually either don't have a link on top of the image, or link to a gallery with all the images in the article. There wouldn't be a gallery in this case.

If the goal is to show an even bigger image then we'd need to have two different images for every uploaded one (basically the original and a scaled one) as @benhalpern mentioned here https://github.com/thepracticaldev/dev.to/issues/413#issuecomment-415172967

Last but not least, if you decide to include an image viewer component, it has to be as small as possible :)

**Comment by deadcoder0904** [comment]
> I don't see much advantage

The advantage is sometimes nobody can read the contents of the image. So it's weird to open the image in the same tab (since Firefox's View Image opens in same tab) then zoom the image in, read all its contents & go back (which sends another request).

The advantage is pretty obvious IMO.

**Comment by rhymes** [comment]
@deadcoder0904 I get what you're saying, it wasn't obvious to me because I always Meta-click on things so they open in a new tab, but that's how my brain is programmed :D 

Anyway, I think if done well, why not :) 

ps. it should also be super easy to close the overlay, like clicking outside of it of pressing ESC or something

**Comment by araslanov-e** [comment]
https://github.com/thepracticaldev/dev.to/pull/3859

**Comment by araslanov-e** [comment]
I think that a link to the image is needed in case js does not work, but you need to show the original image

**Comment by araslanov-e** [comment]
I suggest https://github.com/sparanoid/lightense-images

**Comment by Link2Twenty** [comment]
lightense-images seems to work quite nicely. I've written a quick console bodge for anyone that wants to see it in action.

```js
// inject lightense code
const tag = document.createElement('script');
tag.src = "https://cdn.jsdelivr.net/gh/sparanoid/lightense-images@master/lightense.min.js";
tag.addEventListener('load', ()=>{
  // prevent image clicks opening the link
  document.querySelectorAll(".article-body-image-wrapper").forEach(dom => dom.addEventListener('click', e=>e.preventDefault()));
  // run lightense on all images
  Lightense('.article-body-image-wrapper img');
  console.log('ready!')
});
document.body.appendChild(tag);
```
![Forum_Lightbox](https://user-images.githubusercontent.com/3534427/91171239-0f27ab00-e6d2-11ea-8410-ee892d04ecbd.gif)

**Comment by Link2Twenty** [comment]
Also they use CSS variables so this can be hooked up to the forem theming system

```css
--lightense-z-index: ;
--lightense-backdrop: ;
--lightense-duration: ;
--lightense-timing-func: ;
```

I've created a pull request to expose `backdrop-filter` if they take this we might be able to have a little think about what else we'd like to expose.

https://github.com/sparanoid/lightense-images/pull/51

For instance perhaps we'd like to be able to add a drop shadow to images when they're open.

![dropshadow example](https://user-images.githubusercontent.com/3534427/91193019-8e2adc80-e6ee-11ea-88df-685573e02fd4.png)

**Comment by Link2Twenty** [comment]
I went back and checked to see what was happening with the pull request and noticed this...

![banned user](https://user-images.githubusercontent.com/3534427/93318903-248e8180-f807-11ea-8617-ce59c19ad730.png)

I guess now we have a couple of options, look for another project we can use or fork lightense, which has an MIT licence, to bring it under the Forem umbrella and maintain it ourselves.

I guess option 2 would need an ok from @benhalpern though.

**Comment by cmgorton** [comment]
Since this issue has been brought up several times I am going to add it to our feature backlog project to get more visibility on it.

**Comment by yoriiis** [comment]
Hello, this discussion is old, but the zoom feature for images in articles is still not available. Are you currently working on this, or is it still a work in progress?

It's quite annoying not to be able to display an image in large format. Currently, you are redirected to the file itself and the size has been resized, so it is just as small as in the article.

Thanks
